word_frequency_java/WordFrequency.java

```diff
@@ -0,0 +1,51 @@
+import java.util.StringTokenizer;
+import java.util.HashMap;
+
+public class WordFrequency{
+
+    HashMap<String, Integer> wordFrequency;
+
+    public WordFrequency(){
+        wordFrequency = new HashMap<String, Integer>();
+    }
+
+    public void buildWordFrequencyTable(String words){
+        StringTokenizer st = new StringTokenizer(words);
+        String next;
+        while (st.hasMoreTokens()){
+            next = st.nextToken();
+            // If not a space and is a valid word add to
+            // hashmap and increment value appropriately
+            if (!next.equals(" ") && isValidWord(next)){
+                if (wordFrequency.containsKey(next)){
+                    wordFrequency.put(next, wordFrequency.get(next) + 1);
+                }else{
+                    wordFrequency.put(next, 1);
+                }
+            }
+        }
+    }
+
+    public void printTable(){
+        System.out.println("\t Word \t Frequency");
+        int totalWords = 0;
+        for ( String word : wordFrequency.keySet() ){
+            System.out.println("\t " + word + " \t " + wordFrequency.get(word));
+            totalWords += wordFrequency.get(word);
+        }
+        System.out.println("Total Words in the table : " + totalWords);
+    }
+
+
+    public static boolean isValidWord(String word){
+        for (char c: word.toCharArray()){
+            int ascii = c;
+            if ( (ascii < 97 || ascii > 122) &&
+                 (ascii < 65 || ascii > 90) 
```

Comment:
For better readability, most languages allow something like this:
`if ( (c < 'a' || ascii > 'Z' ) && ( c < 'A' || ascii > 'Z') `
[you might have to cast to int]

Reply:
Good call, changed